components/faction/commons/faction.lua

```diff
@@ -60,36 +60,48 @@ local byLowerName = Table.mapValues(byName,
 )
 
 ---Returns a list of valid factions
----@param game string?
+---@param options {game: string?}?
 ---@return string[]
-function Faction.getFactions(game)
-	game = game or Data.defaultGame
+function Faction.getFactions(options)
+	if type(options) ~= 'table' then
```

Comment:
am on mobile, hence can not span across several lines, comment is intended to be for lines 66-69

since we use this 4 times, maybe make it a function
```lua
function Faction._getGame(options)
	return (type(options) == 'table' and options or {}).game or Data.defaultGame
end
```
could then do `local game = Faction._getGame(options)`
(not sure about the function name though)

Reply:
Maybe _parseOptions and return options with potential defaultGame set?

Reply:
sounds okay